src/wallet/wallet.cpp

```diff
@@ -210,9 +210,11 @@ WalletCreationStatus CreateWallet(interfaces::Chain& chain, const SecureString&
             }
 
             // Set a seed for the wallet
-            CPubKey master_pub_key = wallet->m_spk_man->GenerateNewSeed();
-            wallet->m_spk_man->SetHDSeed(master_pub_key);
-            wallet->m_spk_man->NewKeyPool();
+            {
+                if (auto spk_man = wallet->m_spk_man.get()) {
+                    spk_man->SetupGeneration();
```

Comment:
In commit "Refactor: Move SetupGeneration code out of CWallet" (f95750d52a21a69ad55db8422f10c563fb9e08ac)

Should this be checking the return code from SetupGeneration and setting an error if it fails? Old code wasn't checking the return value from NewKeyPool either, but that doesn't seem right.

Reply:
It should. Done.